src/library/scala/collection/immutable/ListMap.scala

```diff
@@ -189,8 +217,55 @@ sealed class ListMap[A, +B] extends AbstractMap[A, B]
       else removeInternal(k, cur.next, cur :: acc)
 
     override protected def next: ListMap[A, B1] = ListMap.this
+    @inline private[ListMap] def next0 = next
```

Comment:
yes - its called by the companion, which  cant call a protected method

Reply:
That makes sense, although we could make that method an instance method? Anyway this way works too

Reply:
It is deliberately not an instance method as it is not operating on an instance

Reply:
I see your reasoning. Would it better to give it a clearer name and do casting here instead of callsite? e.g. `@inline private[ListMap] def nextNode = next.asInstanceOf[this.type]`

Reply:
Your cast is not accurate as the generic type is different (Its visibility, not type that is the issue)
I tend to use the suffix0/1 for associated implementation detail. This pattern is common across the java and scala codebase

Reply:
> yes - its called by the companion, which cant call a protected method

Companions can access private and protected methods, so I don't see why this would be necessary?